src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Range.cs

```diff
@@ -0,0 +1,177 @@
+            ArrayBuilder<LocalSymbol> locals = ArrayBuilder<LocalSymbol>.GetInstance();
+            ArrayBuilder<BoundExpression> arguments = ArrayBuilder<BoundExpression>.GetInstance();
+
+            // left.HasValue && right.HasValue
+            BoundExpression condition = null;
+            foreach (var operand in operands)
+            {
+                if (operand.Type.IsNullableType())
+                {
+                    BoundExpression tempOperand = CaptureExpressionInTempIfNeeded(operand, sideeffects, locals);
+                    BoundExpression operandHasValue = MakeOptimizedHasValue(tempOperand.Syntax, tempOperand);
+
+                    if (condition is null)
+                    {
+                        condition = operandHasValue;
+                    }
+                    else
+                    {
+                        // PROTOTYPE: make sure this type exists in binding
+                        TypeSymbol boolType = _compilation.GetSpecialType(SpecialType.System_Boolean);
+                        condition = MakeBinaryOperator(syntax, BinaryOperatorKind.BoolAnd, condition, operandHasValue, boolType, method: null);
+                    }
+
+                    arguments.Add(MakeOptimizedGetValueOrDefault(tempOperand.Syntax, tempOperand));
+                }
+                else
+                {
+                    arguments.Add(operand);
+                }
+            }
+
+            Debug.Assert(condition != null);
+
+            // method(left.GetValueOrDefault(), right.GetValueOrDefault())
+            BoundExpression rangeCall = MakeCall(syntax, rewrittenReceiver: null, method, arguments.ToImmutableAndFree(), type);
+
+            // new Nullable(method(left.GetValueOrDefault(), right.GetValueOrDefault()))
+            // PROTOTYPE: make sure this ctor exists in binding
+            MethodSymbol nullableCtor = (MethodSymbol)_compilation.GetSpecialTypeMember(SpecialMember.System_Nullable_T__ctor).SymbolAsMember((NamedTypeSymbol)type);
+            BoundExpression consequence = new BoundObjectCreationExpression(syntax, nullableCtor, binderOpt: null, rangeCall);
+
+            // default
+            BoundExpression alternative = new BoundDefaultExpression(syntax, constantValueOpt: null, type);
+
+            // left.HasValue && right.HasValue ? new Nullable(method(left.GetValueOrDefault(), right.GetValueOrDefault())) : default
+            BoundExpression conditionalExpression = RewriteConditionalOperator(
+                syntax: syntax,
+                rewrittenCondition: condition,
+                rewrittenConsequence: consequence,
+                rewrittenAlternative: alternative,
+                constantValueOpt: null,
+                rewrittenType: type,
+                isRef: false);
+
+            return new BoundSequence(
+                syntax: syntax,
+                locals: locals.ToImmutableAndFree(),
```

Comment:
>locals: locals.ToImmutableAndFree(), [](start = 16, length = 36)

Because of this, lifted range operators should probably be disallowed in an expression tree context.